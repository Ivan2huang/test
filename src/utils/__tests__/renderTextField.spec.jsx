import React from 'react';
import { render } from '@testing-library/react';

import withIntl from '../../i18n/withIntlProvider';
import RenderTextField from '../renderTextField';

describe('render Text Field utils', () => {
  it('should match snapshot', () => {
    const Component = withIntl(RenderTextField);
    const { container } = render(
      <Component
        label="test-label"
        name="test"
        value="test"
        testId="testID"
        helperText="helperText"
        input={{
          value: 'test',
          onChange: jest.fn(),
        }}
        meta={{
          invalid: false,
          touched: false,
          error: false,
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when error without error message', () => {
    const Component = withIntl(RenderTextField);
    const { container } = render(
      <Component
        label="test-label"
        name="test"
        value="test"
        testId="testID"
        helperText="helperText"
        input={{
          value: 'test',
          onChange: jest.fn(),
        }}
        meta={{
          invalid: true,
          touched: true,
          error: true,
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when error with error message', () => {
    const Component = withIntl(RenderTextField);
    const { container } = render(
      <Component
        label="test-label"
        name="test"
        value="test"
        testId="testID"
        helperText="helperText"
        errorMessage="error"
        input={{
          value: 'test',
          onChange: jest.fn(),
        }}
        meta={{
          invalid: true,
          touched: true,
          error: true,
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when error is a string', () => {
    const Component = withIntl(RenderTextField);
    const { container } = render(
      <Component
        label="test-label"
        name="test"
        value="test"
        testId="testID"
        helperText="helperText"
        input={{
          value: 'test',
          onChange: jest.fn(),
        }}
        meta={{
          invalid: true,
          touched: true,
          error: 'Invalid value',
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when error is a locale object', () => {
    const Component = withIntl(RenderTextField);
    const { container } = render(
      <Component
        label="test-label"
        name="test"
        value="test"
        testId="testID"
        helperText="helperText"
        input={{
          value: 'test',
          onChange: jest.fn(),
        }}
        meta={{
          invalid: true,
          touched: true,
          error: {
            id: 'error.message.id',
            defaultMessage: 'Error message default',
          },
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
