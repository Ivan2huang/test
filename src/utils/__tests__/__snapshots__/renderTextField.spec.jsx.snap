// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Text Field utils should match snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
    data-testid="testID"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink="true"
      for="input-undefined"
    >
      test-label
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      label="test-label"
    >
      <input
        aria-describedby="helptext-undefined"
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        id="input-undefined"
        type="text"
        value="test"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-59 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-60"
          style="width: 0px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      aria-live="polite"
    >
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
        id="helptext-undefined"
      >
        helperText
      </p>
    </div>
  </div>
</div>
`;

exports[`render Text Field utils should match snapshot when error is a locale object 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
    data-testid="testID"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-error Mui-error MuiFormLabel-filled"
      data-shrink="true"
      for="input-undefined"
    >
      test-label
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
      label="test-label"
    >
      <input
        aria-describedby="helptext-undefined"
        aria-invalid="true"
        class="MuiInputBase-input MuiOutlinedInput-input"
        id="input-undefined"
        type="text"
        value="test"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-323 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-324"
          style="width: 0px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      aria-live="polite"
    >
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error MuiFormHelperText-filled"
        id="helptext-undefined"
      >
        Error message default
      </p>
    </div>
  </div>
</div>
`;

exports[`render Text Field utils should match snapshot when error is a string 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
    data-testid="testID"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-error Mui-error MuiFormLabel-filled"
      data-shrink="true"
      for="input-undefined"
    >
      test-label
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
      label="test-label"
    >
      <input
        aria-describedby="helptext-undefined"
        aria-invalid="true"
        class="MuiInputBase-input MuiOutlinedInput-input"
        id="input-undefined"
        type="text"
        value="test"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-255 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-256"
          style="width: 0px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      aria-live="polite"
    >
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error MuiFormHelperText-filled"
        id="helptext-undefined"
      >
        Invalid value
      </p>
    </div>
  </div>
</div>
`;

exports[`render Text Field utils should match snapshot when error with error message 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
    data-testid="testID"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-error Mui-error MuiFormLabel-filled"
      data-shrink="true"
      for="input-undefined"
    >
      test-label
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
      label="test-label"
    >
      <input
        aria-describedby="helptext-undefined"
        aria-invalid="true"
        class="MuiInputBase-input MuiOutlinedInput-input"
        id="input-undefined"
        type="text"
        value="test"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-187 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-188"
          style="width: 0px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      aria-live="polite"
    >
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error MuiFormHelperText-filled"
        id="helptext-undefined"
      >
        error
      </p>
    </div>
  </div>
</div>
`;

exports[`render Text Field utils should match snapshot when error without error message 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
    data-testid="testID"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-error Mui-error MuiFormLabel-filled"
      data-shrink="true"
      for="input-undefined"
    >
      test-label
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
      label="test-label"
    >
      <input
        aria-describedby="helptext-undefined"
        aria-invalid="true"
        class="MuiInputBase-input MuiOutlinedInput-input"
        id="input-undefined"
        type="text"
        value="test"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-127 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-128"
          style="width: 0px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      aria-live="polite"
    />
  </div>
</div>
`;
